lib/Frontend/CachingUtils.cpp

```diff
@@ -281,7 +284,17 @@ bool replayCachedCompilerOutputs(
                     toString(File.takeError()));
       continue;
     }
-    *File << Output.Proxy.getData();
+
+    if (UseCASBackend && Output.Kind == file_types::ID::TY_Object) {
+      auto Schema = std::make_unique<llvm::mccasformats::v1::MCSchema>(CAS);
+      SmallString<50> ContentsStorage;
```

Comment:
Where does the `50` come from?

Reply:
I stole this from `llvm-project/clang/lib/Frontend/CompileJobCache.cpp:731` where we also serialize the object file from MCCAS, didn't think about the default allocation size. Is it better to leave this blank?

```
SmallString<50> ContentsStorage;
    if (IsOutputFile && ComputedJobNeedsReplay) {
      llvm::raw_svector_ostream OS(ContentsStorage);
      if (WriteOutputAsCASID)
        llvm::cas::writeCASIDBuffer(CAS.getID(O.Object), OS);
      else if (UseCASBackend) {
        // Replay by write out object file.
        // When the environmental variable is set, save the backend CASID for
        // analysis later.
        if (llvm::sys::Process::GetEnv("CLANG_CAS_BACKEND_SAVE_CASID_FILE")) {
          std::string CASIDPath = Path + ".casid";
          std::error_code EC;
          llvm::raw_fd_ostream IDOS(CASIDPath, EC);
          if (EC)
            return llvm::errorCodeToError(EC);
          writeCASIDBuffer(CAS.getID(O.Object), IDOS);
        }
        auto Schema = std::make_unique<llvm::mccasformats::v1::MCSchema>(CAS);
        if (auto E = Schema->serializeObjectFile(*Obj, OS))
          return E;
      }
```